worker/tests/common.go

```diff
@@ -46,6 +46,29 @@ func TestWorkerExec(t *testing.T, w *base.Worker) {
 
 	id := identity.NewID()
 
+	// verify pid1 exits when stdin sees EOF
+	stdin := bytes.NewReader([]byte("hello"))
```

Comment:
could you add a case where instead of using `bytes.NewReader` this is a pipe and only after `started` channel has returned you write into stdin and then close it.

Reply:
Updated to use `io.Pipe`